src/protocols/bdx/BdxTransferSession.cpp

```diff
@@ -64,82 +59,50 @@ TransferSession::TransferSession()
     mSuppportedXferOpts.ClearAll();
 }
 
-void TransferSession::PollOutput(OutputEvent & event, System::Clock::Timestamp curTime)
+void TransferSession::DispatchOutputEvent(TransferSession::OutputEvent & outputEvent)
```

Comment:
This significantly changes the public API of BDX transfer session. PollOutput doesn't mean that it actually polls. It is a misnomer. PollOutput just means "give me the next action to take". Triggering that from messages is what was needed. This change of public API completely changes the directionality of the transfer session API. This will break external clients that were not using the TransferFacilitator.